pkg/collector/corechecks/system/cpu/cpu/cpu.go

```diff
@@ -23,57 +23,99 @@ import (
 const CheckName = "cpu"
 
 // For testing purpose
-var cpuTimesFunc = cpu.Times
-var cpuInfoFunc = cpu.Info
+var getCpuTimes = cpu.Times
+var getCpuInfo = cpu.Info
+var getContextSwitches = GetContextSwitches
 
 // Check doesn't need additional fields
 type Check struct {
 	core.CheckBase
-	nbCPU       float64
 	lastNbCycle float64
 	lastTimes   cpu.TimesStat
 }
 
 // Run executes the check
 func (c *Check) Run() error {
+	log.Debug("running cpu check")
```

Comment:
This could already be done by the base Check class? I'm not sure..

Reply:
What do you mean? We're not calling the base class here

Reply:
I'm sorry, I meant at the scheduler level. We're probably (definitely?) already logging running checks.

Reply:
```
2024-12-16 15:43:54 +07 | CORE | INFO | (pkg/collector/worker/check_logger.go:40 in CheckStarted) | check:cpu | Running check...
2024-12-16 15:43:54 +07 | CORE | INFO | (pkg/collector/worker/check_logger.go:59 in CheckFinished) | check:cpu | Done running check
```

Reply:
Removed!